flow/project.py

```diff
@@ -840,33 +859,36 @@ def _resolve_directives(self, name, defaults, env):
             all_directives.update(defaults.get(name, {}))
         return all_directives
 
-    def _submit_cmd(self, entrypoint, ignore_conditions, jobs=None):
+    def _submit_cmd(self, entrypoint, ignore_conditions, jobs):
         entrypoint = self._determine_entrypoint(entrypoint, {}, jobs)
         cmd = f"{entrypoint} run -o {self.name}"
         cmd = cmd if jobs is None else cmd + f" -j {get_aggregate_id(jobs)}"
-        cmd = cmd if self.options is None else cmd + " " + self.options
+        options = self.submit_options
         if ignore_conditions != IgnoreConditions.NONE:
-            return cmd.strip() + " --ignore-conditions=" + str(ignore_conditions)
-        return cmd.strip()
+            options += " --ignore-conditions=" + str(ignore_conditions)
+        return " ".join((cmd, options)).strip()
 
     def _run_cmd(self, entrypoint, operation_name, operation, directives, jobs):
         if isinstance(operation, FlowCmdOperation):
             return operation(*jobs).lstrip()
         entrypoint = self._determine_entrypoint(entrypoint, directives, jobs)
-        return f"{entrypoint} exec {operation_name} {get_aggregate_id(jobs)}".lstrip()
+        cmd = f"{entrypoint} exec {operation_name} {get_aggregate_id(jobs)} {self.run_options}"
+        return cmd.strip()
 
     def __iter__(self):
         yield from self.operations.values()
 
     def __repr__(self):
         return (
             "{type}(name='{name}', operations='{operations}', "
-            "operation_directives={directives}, options='{options}')".format(
+            "operation_directives={directives}, submit_options='{submit_options}', "
+            "run_options='{run_options}')".format(
```

Comment:
Generally you want to call `repr` to generate representations of data when it's used in a `__repr__`. For example, the single quotes wrapping `'{submit_options}'` won't be correct if the string itself contains single quotes. Using `repr` will handle quoting/escaping the string correctly for you. This example is kind of unusual because we don't generate a repr that is literally interpretable -- the string list of operations breaks that.

We probably ought to replace `name='{name'}` with `name={repr(name)}` and similar for the other string arguments like `submit_options`, `run_options`.